streamlit_app.py

```diff
@@ -104,6 +104,39 @@ def parse_nada_json(json_data):
     
     return input_info, output_info
 
+import streamlit as st
+
+def create_party_inputs(input_info, input_values):
+    party_names = sorted(set(info['party'] for info in input_info.values()))
+    updated_input_values = input_values.copy()
+
+    if len(party_names) > 1:
+        # Create two columns if there's more than one party
+        columns = st.columns(2)
+    else:
+        # Create a single column if there's only one party
+        columns = [st.columns(1)[0]]
```

Comment:
What are these columns?

Reply:
these are 2 columns to condense the space taken up by inputs if there are inputs coming from multiple parties (if there are 10 parties each providing 1 input, this splits them visually into 2 columns with 5 each)